lib/redix/pubsub/connection.ex

```diff
@@ -162,6 +162,18 @@ defmodule Redix.PubSub.Connection do
     end
   end
 
+
+  # Ask Redix.Utils to setup a connection, then configure the returned
+  # socket with http://erlang.org/doc/man/inet.html#setopts-2
+  #
+  @spec establish_connection(Keyword.t) :: {:ok, :gen_tcp.socket} | {:error, term} | {:stop, term}
```

Comment:
I wasn't sure about this. I googled and found [this comment](https://github.com/elixir-lang/elixir/issues/3144#issuecomment-77678584), but as you say I'm not certain it's widely used in the wild.

I wanted to annotate this because I think that the spec for `Redix.Utils.connect/1` is not correct but I'm not really sure:

In both versions it says that `{:stop, term, %{}}` can be returned, but I don't see it in the code.
https://github.com/whatyouhide/redix/blob/v0.5.2/lib/redix/utils.ex#L61
https://github.com/whatyouhide/redix/blob/v0.6.0/lib/redix/utils.ex#L61

Rather than leaving it undefined, I preferred to put a precise annotation here based on what's expected by the consumers.